### DISCUSSION TITLE
Differentiate blocks in the docs for blockheads like me

### DISCUSSION THREAD

**Discussion by geoidesic** [zotonic/zotonic#3955] [General]
There seem to be two types of block: template blocks and resource blocks. This isn't made clear in the docs. 

## Template blocks 
- are discussed here: https://zotonic.com/docs/1346/block
- and here: https://zotonic.com/cookbook/1583/how-to-add-a-custom-content-block

I feel like these pages should be connected, since they cover the same ground?

## Resource blocks
- are nowhere to be found in the docs: they should be in the User Guide. 

## Questions 
(about resource blocks)
1. Why is the `standard` type of Block greyed out?
2. What are the other types for?

**Comment by mworrell** [comment]
The `standard` is greyed out because it is a section header for the kind of blocks.
There can be multiple sections of block-types, depending on modules adding them.

<img width="199" alt="Screenshot 2024-12-02 at 14 58 21" src="https://github.com/user-attachments/assets/ee60844d-6ec5-4be2-a6f5-ff6a9bd70d41">

The standard types are:

 * Header - used to make a big header
 * Text - used to add a (rich) text block (using TinyMCE)
 * Embed page - to add some citation / popup / link to another page

The representation depends on your templates.  The default templates are very simple.

The header is there for surveys, where you often just want as simple header between different questions.


And indeed, maybe these "blocks" could be called differently, or just get a little extra word.  So that there is a difference between content-blocks and template-blocks.

**Reply by geoidesic** [reply]
I suggest `standard` should be explained, somewhere. It's cryptic as it stands.
All four options should be explained in the User Manual (which is currently empty)
Yes, please consider differentiating template-blocks and content-blocks – getting the DSL (domain specific language) right (i.e. consistent and unambiguous) for Zotonic is crucial to enabling new users / devs to grok it quickly. It's also crucial for enabling AI assisted coding to work well. The current idiom is to add the docs to your AI IDE so that the IDE understands the framework. If the docs are missing or confusing then that cuts Zotonic out of the AI coding race.

**Reply by mworrell** [reply]
We are a bit stuck with `blocks` due to the resource property being called `blocks` (and all templates depending on them).

Maybe if we call them `page blocks`  and `template blocks` ?


About feeding the docs to a LLM (AI)... could we feed that AI a website?  As we prefer to maintain the docs there and not in the RST format.

**Reply by mworrell** [reply]
`Standard` might be `Default Zotonic page blocks` -- more words might help here.

Also adding a description to each might help.

**Reply by geoidesic** [reply]
Yes, a website is fine as a source for AI to read docs.

**Reply by mworrell** [reply]
Do you have any opinion about the naming of the two kinds of blocks?

**Reply by geoidesic** [reply]
I think what's important is that they are differentiated in the documentation. "Template block" is a good name, to keep parity with django cms. 

It might be better to have a completely different name for what's in the page cms editor, e.g. page regions or page sections. 

But if that's not possible then yes, "page blocks" works.

**Reply by mworrell** [reply]
Yes, "page sections" also came to mind.

And then document in the developer documentation that a "page section" is called "blocks" under the hood.

That is similar to how in the admin we talk about "pages" and "connections", when in the code it is "resources" and "edges."

**Reply by mworrell** [reply]
We have a merge request with the textual changes in the admin here: https://github.com/zotonic/zotonic/pull/3975
